showyourwork/workflow/rules/preprocess.smk

```diff
@@ -56,7 +58,7 @@ rule:
             --keep-logs           \\
             --keep-intermediates  \\
             -r 0                  \\
-            "{input[0]}"
+            "{input[0]}" {joined_user_args}
```

Comment:
User args should be before `input`.

Reply:
I thought so too! But in tectonic they actually go after

Reply:
It's weird. I'm doing a sanity check right now and looks like there are examples of both in the docs and in user issues... And when I initially tried putting the `-Z shell-escape` before the input, it failed. Very strange.

Reply:
The old behavior had them first: https://github.com/showyourwork/showyourwork/blob/a3d8b337ec1390eb021e63377643f79bc8d01783/showyourwork/tex.py#L62

Reply:
I'm about 99% sure that the syntax as written isn't adding any arguments, so I'm surprised you're finding it to work at all!

Reply:
Sometimes the 1% wins...
![image](https://user-images.githubusercontent.com/7593028/235135241-d8ab8a4c-07a7-4957-84f7-6c812d291642.png)

![image](https://user-images.githubusercontent.com/7593028/235135033-1754273b-f3b7-4a15-9b8d-c9e256b734ee.png)

Very weird!

Reply:
Anyways I will definitely change it, not to worry. I'm just marveling at why it works...

Reply:
For posterity:

![image](https://user-images.githubusercontent.com/7593028/235135603-436c764d-46b8-4a9f-b0a4-e45b70847f38.png)

So looks like tectonic treats its arguments as an unordered set...

Reply:
The 99% was about the snakemake access to globals, I'm never confident about anything tectonic related :). I'm worried that you're seeing it working because of a caching issue rather than the new command working, but I could be wrong.

Reply:
I did a `showyourwork clean` in between so that's probably it!